lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3OutputStream.java

```diff
@@ -329,6 +338,11 @@ private void finishUpload(String uploadId)
                 .key(location.key())
                 .uploadId(uploadId)
                 .multipartUpload(x -> x.parts(parts))
+                .applyMutation(builder -> {
```

Comment:
We need to handle the exclusive create failure when doing a multi-part upload.

Reply:
the conflict exception is thrown on the completion according to the docs: 

![412_conwrite](https://github.com/user-attachments/assets/ad45b18b-6c15-4dab-bce6-fe81c9a7fbc4)

Reply:
This is handled by the 
```
try {
            uploadId.ifPresent(this::finishUpload);
        }
        catch (SdkException e) {
            abortUploadSuppressed(e);
            throw new IOException(e);
        }
```

Reply:
What else do you have in mind @electrum ?